ivadomed/loader/loader.py

```diff
@@ -297,8 +307,18 @@ def _load_filenames(self):
 
             input_data_shape, _ = seg_pair.get_pair_shapes()
 
+            resample = False
+            for idx, transfo in enumerate(self.transform.transform["im"].transforms):
+                if "Resample" in str(type(transfo)):
+                    resample = True
+                    resample_param = (transfo.hspace, transfo.wspace, transfo.dspace)
+
             for idx_pair_slice in range(input_data_shape[-1]):
                 slice_seg_pair = seg_pair.get_pair_slice(idx_pair_slice)
+                self.bounding_box = 'bounding_box' in slice_seg_pair['input_metadata'][0]
```

Comment:
I am not sure how to fix this but.. here we assume that all images of the dataset has bounding_box in its metadata. If not, it could (if unlucky) crash later.

Reply:
maybe @Drulex has some ideas?

Reply:
Some general comments without digging too deep:
1. `bounding_box` -> `has_bounding_box` or something more representative since `bounding_box` could mean an actual box/coordinates of a box (this is what I thought at first).
2. If `bounding_box` is local to this function/iteration it should not be a class attribute. I am confused that `self.bounding_box` is set in each iteration of this loop, but later referenced in `__get_item()__`. Is it supposed to be a global flag to ensure that all `slice_seg_pair` have a `bounding_box`? In that case it is better to do something like this:

```
# in __init__() have self.has_bounding_box = True
# [...]
for idx_pair_slice in range(input_data_shape[-1]):
    slice_has_bounding_box = 'bounding_box' in slice_seg_pair['input_metadata'][0]
    self.has_bounding_box &= slice_has_bounding_box # notice bitwise operator here

# [...] later when we want to know if all images have the bounding box
if not self.has_bounding_box:
    # log appropriate message
    # raise ValueError("Some images are missing bounding_box in their metadata!") or something more useful
```
^ this assumes that if we are ok with a potential `IndexError` if `slice_seg_pair['input_metadata'][0]` does not exist, otherwise it's better to sanitize the data somewhere in the initialization.

Reply:
Fixed it with imed_obj_detect.verify_metadata -> verifies all input_metadata, gt_metadata and indices have the 'bounding_box' key